--- sympy-1.0/doc/Makefile~	2016-03-08 19:38:39.000000000 +0100
+++ sympy-1.0/doc/Makefile	2016-05-10 08:10:27.649406764 +0200
@@ -117,7 +117,7 @@
 	pdflatex -output-directory=_build/cheatsheet cheatsheet/cheatsheet.tex
 	pdflatex -output-directory=_build/cheatsheet cheatsheet/cheatsheet.tex
 
-_build/cheatsheet/cheatsheet.pdf: cheatsheet/combinatoric_cheatsheet.tex
+_build/cheatsheet/combinatoric_cheatsheet.pdf: cheatsheet/combinatoric_cheatsheet.tex
 	mkdir -p _build/cheatsheet
 	pdflatex -output-directory=_build/cheatsheet cheatsheet/combinatoric_cheatsheet.tex
 	pdflatex -output-directory=_build/cheatsheet cheatsheet/combinatoric_cheatsheet.tex
diff -ur sympy-1.1.1.org/doc/api/conf.py sympy-1.1.1/doc/api/conf.py
--- sympy-1.1.1.org/doc/api/conf.py	2017-07-22 00:17:42.000000000 +0200
+++ sympy-1.1.1/doc/api/conf.py	2018-07-04 21:04:02.333475214 +0200
@@ -131,4 +131,4 @@
 #latex_appendices = []
 
 # If false, no module index is generated.
-#latex_use_modindex = True
+#latex_domain_indices = True
diff -ur sympy-1.1.1.org/doc/src/conf.py sympy-1.1.1/doc/src/conf.py
--- sympy-1.1.1.org/doc/src/conf.py	2017-07-27 00:46:51.000000000 +0200
+++ sympy-1.1.1/doc/src/conf.py	2018-07-04 21:03:49.133014108 +0200
@@ -178,7 +178,7 @@
 latex_show_pagerefs = True
 
 # We use False otherwise the module index gets generated twice.
-latex_use_modindex = False
+latex_domain_indices = False
 
 default_role = 'math'
 pngmath_divpng_args = ['-gamma 1.5', '-D 110']
