--- sympy-1.11.1/doc/Makefile.orig	2023-01-04 19:37:36.910055980 +0100
+++ sympy-1.11.1/doc/Makefile	2023-01-04 19:42:23.068505727 +0100
@@ -60,7 +60,6 @@ html:
 # Force colors on CI. Set NO_COLOR=1 on the command line to disable this.
 _html: export FORCE_COLOR = 1
 _html: $(BUILDDIR)/html/pics/*.png
-_html: SPHINXOPTS += -W --keep-going
 _html: $(BUILDDIR)/logo/sympy-notailtext-favicon.ico
 	mkdir -p $(SOURCEDIR)/.static
 	mkdir -p $(BUILDDIR)/html
