--- sympy-1.5.1/sympy/utilities/runtests.py.orig	2019-12-21 00:38:01.000000000 +0100
+++ sympy-1.5.1/sympy/utilities/runtests.py	2020-04-08 21:39:01.146014778 +0200
@@ -524,8 +524,10 @@
     split = kwargs.get('split', None)
     time_balance = kwargs.get('time_balance', True)
     blacklist = kwargs.get('blacklist', ['sympy/integrals/rubi/rubi_tests/tests'])
-    if ON_TRAVIS:
+    if ON_TRAVIS or os.getenv('DISPLAY', '') == '':
         # pyglet does not work on Travis
+        blacklist.extend('sympy/categories/tests/test_drawing.py')
+        blacklist.extend('sympy/printing/tests/test_preview.py')
         blacklist.extend(['sympy/plotting/pygletplot/tests'])
     blacklist = convert_to_native_paths(blacklist)
     fast_threshold = kwargs.get('fast_threshold', None)
@@ -702,7 +704,9 @@
             import matplotlib
             matplotlib.use('Agg')
 
-    if ON_TRAVIS or import_module('pyglet') is None:
+    if ON_TRAVIS or os.getenv('DISPLAY', '') == '' or import_module('pyglet') is None:
+        blacklist.extend(["sympy/categories/diagram_drawing.py"])
+        blacklist.extend(["sympy/printing/preview.py"])
         blacklist.extend(["sympy/plotting/pygletplot"])
 
     if import_module('theano') is None:
